swarm.sh

```diff
@@ -1,79 +1,147 @@
 #!/bin/bash
 
-# Constants
-readonly ACTION=$1
-readonly MODE=$2
-
-COMPOSE_FILE_PATH=$(
-  cd "$(dirname "${BASH_SOURCE[0]}")" || exit
-  pwd -P
-)
-readonly COMPOSE_FILE_PATH
-
-# Import libraries
-ROOT_PATH="${COMPOSE_FILE_PATH}/.."
-. "${ROOT_PATH}/utils/config-utils.sh"
-. "${ROOT_PATH}/utils/docker-utils.sh"
-. "${ROOT_PATH}/utils/log.sh"
+declare ACTION=""
+declare MODE=""
+declare COMPOSE_FILE_PATH=""
+declare UTILS_PATH=""
+declare SERVICE_IMPORTER_NAMES=()
+declare SERVICE_NAMES=()
+declare ALL_SERVICES=()
+
+function init_vars() {
+  ACTION=$1
+  MODE=$2
+
+  COMPOSE_FILE_PATH=$(
+    cd "$(dirname "${BASH_SOURCE[0]}")" || exit
+    pwd -P
+  )
+
+  UTILS_PATH="${COMPOSE_FILE_PATH}/../utils"
+
+  SERVICE_IMPORTER_NAMES=(
+    "disi-jsreport-config-importer"
+    "disi-es-index-importer"
+    "disi-openhim-config-importer"
+    "disi-kibana-config-importer"
+    "message-bus-kafka-config-importer"
+    "sante-mpi-config-importer"
+    "hapi-fhir-config-importer"
+  )
+  SERVICE_NAMES=(
+    "data-mapper-logstash"
+    "mpi-updater"
+    "mpi-checker"
+    "reprocess-mediator"
+  )
+  ALL_SERVICES=(
+    "${SERVICE_NAMES[@]}"
+    "${SERVICE_IMPORTER_NAMES[@]}"
+  )
+
+  readonly ACTION
+  readonly MODE
+  readonly COMPOSE_FILE_PATH
+  readonly UTILS_PATH
+  readonly SERVICE_NAMES
+  readonly SERVICE_IMPORTER_NAMES
+  readonly ALL_SERVICES
+}
+
+# shellcheck disable=SC1091
+function import_sources() {
+  source "${UTILS_PATH}/docker-utils.sh"
+  source "${UTILS_PATH}/config-utils.sh"
+  source "${UTILS_PATH}/log.sh"
+}
+
+function restart_hapi_fhir() {
+  if docker service ps -q instant_hapi-fhir &>/dev/null; then
+    log info "Restarting HAPI FHIR.."
+    try \
+      "docker service scale instant_hapi-fhir=0" \
+      catch \
```

Comment:
Should we not throw the error here? Isn't we want the execution of swarm script to stop when there is an error

Reply:
I think we should add it here. If the restart does not happen, the fhir validation against the IG will not happen

Reply:
cool done :)